src/diffusers/pipelines/pipeline_loading_utils.py

```diff
@@ -749,3 +750,157 @@ def _fetch_class_library_tuple(module):
     class_name = not_compiled_module.__class__.__name__
 
     return (library, class_name)
+
+
+def _identify_model_variants(folder: str, variant: str, config: dict) -> dict:
+    model_variants = {}
+    if variant is not None:
+        for folder in os.listdir(folder):
+            folder_path = os.path.join(folder, folder)
+            is_folder = os.path.isdir(folder_path) and folder in config
+            variant_exists = is_folder and any(p.split(".")[1].startswith(variant) for p in os.listdir(folder_path))
+            if variant_exists:
+                model_variants[folder] = variant
+    return model_variants
+
+
+def _determine_pipeline_class(
+    cls,
+    folder: str,
+    cache_dir: str,
+    config: dict,
+    custom_revision: str,
+    custom_pipeline: str,
+    load_connected_pipeline: bool,
+) -> tuple:
+    custom_class_name = None
+    if os.path.isfile(os.path.join(folder, f"{custom_pipeline}.py")):
+        custom_pipeline = os.path.join(folder, f"{custom_pipeline}.py")
+    elif isinstance(config["_class_name"], (list, tuple)) and os.path.isfile(
+        os.path.join(folder, f"{config['_class_name'][0]}.py")
+    ):
+        custom_pipeline = os.path.join(folder, f"{config['_class_name'][0]}.py")
+        custom_class_name = config["_class_name"][1]
+
+    pipeline_class = _get_pipeline_class(
+        cls,
+        config,
+        load_connected_pipeline=load_connected_pipeline,
+        custom_pipeline=custom_pipeline,
+        class_name=custom_class_name,
+        cache_dir=cache_dir,
+        revision=custom_revision,
+    )
+    return custom_pipeline, pipeline_class
+
+
+def _maybe_raise_warning_for_inpainting(pipeline_class, pretrained_model_name_or_path: str, config: dict):
+    if pipeline_class.__name__ == "StableDiffusionInpaintPipeline" and version.parse(
+        version.parse(config["_diffusers_version"]).base_version
+    ) <= version.parse("0.5.1"):
```

Comment:
Won't this be raised only is diffusers is `<=0.5.1`? Can't we just get rid of it completely?

Reply:
That seems reasonable but I would slightly prefer to keep it even though it's highly unlikely that someone is still using a version that old.

Reply:
Sure. We can remove in v1.0.0